docs/user-guide/pandas.md

```diff
@@ -0,0 +1,156 @@
+`dataset.records()` by wrapping it in a different generator.  
+
+For example, it is possible to automatically flatten all lists with a length of one, and apply this to the `dataset.records()`
+generator as follows:
+
+```python
+def unlist(lst):
+    """
+    If object is a list of length one, return first element. 
+    Otherwise, return original object. 
+    """
+    if isinstance(lst, list) and len(lst) is 1:
+        return lst[0]
+    else:
+        return lst
+
+def dataset_to_pandas(dataset):
+    """
+    Incorporates basic unlisting for easy transfer between Tamr and Pandas. 
+    """ 
+    for record in dataset.records():
+        for key in record:
+            record[key] = unlist(record[key])
+        yield record
+
+df = pd.DataFrame.from_records(dataset_to_pandas(my_dataset))
+```
+
+Similarly, it is possible to filter to extracting only certain attributes, by specifying this in the generator:
+
+```python
+def filter_dataset_to_pandas(dataset, colnames):
+    """
+    Filter the dataset to only the primary key and the columns specified as a list in colnames. 
+    """
+    assert isinstance(colnames, list)
+    colnames = dataset.key_attribute_names + colnames if dataset.key_attribute_names[0] not in colnames else colnames
+    for record in dataset.records():
+        yield {k: unlist(v) for k, v in record.items() if k in colnames}
+
+df = pd.DataFrame.from_records(filter_dataset_to_pandas(my_dataset, ['City', 'new_attr']))
+```
+
+Note that upserting these records would overwite the existing records and attributes, and cause loss of the data 
+stored in the unloaded attributes.  
+
+## Upload Dataframe as Dataset
+
+### Create New Dataset
+To create a new dataset and upload data, the convenience function `dataset.create_from_dataframe()` can be used. 
+Note that Tamr will throw an error if columns aren't generally formatted as strings. (The exception being geospatial
+columns. For that, see the geospatial examples.)
+
+In order to achieve this, the following code will transform the column types to string.
+```python
+df = df.astype(str)
+```
+
+Creating the dataset is as easy as calling:
+```python
+tamr.datasets.create_from_dataframe(df, 'primaryKey', 'my_new_dataset')
+```
+
+### Changing Values
+
+#### Making Changes: In Memory 
+When making changes to a dataset that was loaded as a dataframe, changes can be pushed back to Tamr using the 
+`dataset.upsert_from_dataframe()` method as follows:
+
+```python
+df = pd.DataFrame.from_records(my_dataset.records())
+df['column'] = 'new_value'
+my_dataset.upsert_from_dataframe(df, primary_key_name='primary_key')
+```
+
+#### Making Changes: Streaming
+For larger datasets it might be better to stream the data and apply changes while iterating through the dataset. 
+This way the full dataset does not need to be loaded into memory. 
+```python
+for record in dataset.records():
+    df = pd.DataFrame.from_records(record)
```

Comment:
This call actually isn't working for me with a single record, when I tested it

Reply:
I could add a 
```
   do_something(df)
```
To clarify that it's supposed to be manipulated.